### PR TITLE
[NVPTX] Remove nvvm.bitcast.* intrinsics

### PR DESCRIPTION
**Pull request by AlexMaclean**
Remove the following intrinsics which correspond directly to a bitcast:
- llvm.nvvm.bitcast.f2i
- llvm.nvvm.bitcast.i2f
- llvm.nvvm.bitcast.d2ll
- llvm.nvvm.bitcast.ll2d

### CODE REVIEW

**Comment by Artem-B** [comment]
It may be worth adding a note about this in the release notes.

**Comment by AlexMaclean** [comment]
> It may be worth adding a note about this in the release notes.

I'm not familiar with these, can you point me to an analogous change I could use as an example?

**Comment by justinfargnoli** [comment]
> can you point me to an analogous change I could use as an example?

[[LLD] Add CLASS syntax to SECTIONS](https://github.com/llvm/llvm-project/commit/7e8a9020b1ae3dea28c19f0cd68743482dca13d9#diff-96d3595c06280d8c10f2687758a9e7bb5a0f3552fd53e552a1490411870c3500)
[[LLVM] Add IRNormalizer Pass](https://github.com/llvm/llvm-project/pull/68176/files#diff-a7085286edacbd0cdfabdd7382565889ff52dc9013a847d960d148ff4ad2eff8)

**Review by Artem-B — APPROVED** [review]

File: clang/include/clang/Basic/BuiltinsNVPTX.def [hunk 1]
```diff
@@ -599,14 +599,6 @@ TARGET_BUILTIN(__nvvm_e4m3x2_to_f16x2_rn_relu, "V2hs", "", AND(SM_89,PTX81))
 TARGET_BUILTIN(__nvvm_e5m2x2_to_f16x2_rn, "V2hs", "", AND(SM_89,PTX81))
 TARGET_BUILTIN(__nvvm_e5m2x2_to_f16x2_rn_relu, "V2hs", "", AND(SM_89,PTX81))
 
-// Bitcast
```
Comment (AlexMaclean) [inline clang/include/clang/Basic/BuiltinsNVPTX.def:602]: @jlebar can you confirm it is okay to remove builtins like this? I'm doing this based on your commit 46624a822d3a3df4a4b6dff0d231acb45d269853. Just want to make sure I'm not missing something.

Comment (jlebar) [reply clang/include/clang/Basic/BuiltinsNVPTX.def:602]: (Thank you, past me, for the commit message.)  The key in the commit message from the previous commit is

> These builtins don't need to be backfilled in clang: They're not accessible to user code from nvcc.

If they were accessible from nvcc, we could have compatibility issues where user code (or, just as likely, CUDA headers) expects the builtin to be there.  In which case we might want to keep the builtin (in the same way that we auto-upgrade the llvm intrinsic).

I tried to check using cuda.godbolt.org whether these builtins exist in nvcc, but at this very moment it seems to be down, it's not compiling anything.

Assuming that something like the following does not compile in nvcc, I think this is fine.

```
__global__ void foo() {
  __nvvm_bitcast_f2i(0.0);
}
```

Comment (AlexMaclean) [reply clang/include/clang/Basic/BuiltinsNVPTX.def:602]: Thanks! I've just confirmed these do not work in nvcc.

**Pull request merged**